src/arrayMethodSort.js

```diff
@@ -3,10 +3,29 @@
 /**
  * Implement method Sort
  */
-function applyCustomSort() {
+function applyCustomSort(callback) {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    for (let step = 0; step < this.length; step++) {
+      for (let i = 0; i < this.length - step - 1; i++) {
+        const currValue = this[i];
+        const nextValue = this[i + 1];
+
+        if (nextValue) {
+          if (compareFunction) {
+            if (compareFunction(currValue, nextValue) > 0) {
+              [this[i], this[i + 1]] = [nextValue, currValue];
+            }
+          } else if (nextValue.toString() < currValue.toString()) {
+            [this[i], this[i + 1]] = [nextValue, currValue];
+          }
+        }
+      }
```

Comment:
Try to make it easy. You can do it after setting a default callback (compareFunction).
```suggestion
        if (compareFunction(currValue, nextValue) > 0) {
          [this[i], this[i + 1]] = [nextValue, currValue];
        }
      }
```